cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -122,6 +125,19 @@ public void start() {
       Threads.createDaemonThreadFactory("polling-booking-executor"));
 
     store = storeFactory.create();
+
+    scheduleStore.initialize();
+
+    // Reschedule the persisted schedules
+    synchronized (this) {
```

Comment:
why is this synchronized? To prevent new schedules being added or existing ones being updated/deleted while the service is starting up?